applications/GeoMechanicsApplication/custom_utilities/element_utilities.cpp

```diff
@@ -65,6 +67,27 @@ void GeoElementUtilities::FillPermeabilityMatrix(BoundedMatrix<double, 3, 3>&
     rPermeabilityMatrix(0, 2) = rPermeabilityMatrix(2, 0);
 }
 
+Matrix GeoElementUtilities::FillPermeabilityMatrix(const Element::PropertiesType& Prop, std::size_t Dimension)
+{
+    if (Dimension == 1) {
+        BoundedMatrix<double, 1, 1> result_1d;
+        FillPermeabilityMatrix(result_1d, Prop);
+        return result_1d;
+    }
+    if (Dimension == 2) {
+        BoundedMatrix<double, 2, 2> result_2d;
+        FillPermeabilityMatrix(result_2d, Prop);
+        return result_2d;
+    }
+    if (Dimension == 3) {
+        BoundedMatrix<double, 3, 3> result_3d;
+        FillPermeabilityMatrix(result_3d, Prop);
+        return result_3d;
+    }
+
+    KRATOS_ERROR << "Dimension " << Dimension << " is not supported" << std::endl;
+}
```

Comment:
Looks like a case statement.

Reply:
I had some trouble with the switch for some reason, maybe @avdg81 can help me out next week with this